### PR TITLE
Refactor: initializeDrawerSliders.js

### PR DESCRIPTION
**Pull request by sagarchoudhary96**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description
Refactors file according to eslint rules.

## Related Tickets & Documents
#3777 

# Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

File: app/assets/javascripts/initializers/initializeDrawerSliders.js [hunk 1]
```diff
@@ -1,57 +1,66 @@
...
+  for (let i = 0; i < navLinks.length; i += 1) {
+    navLinks[i].addEventListener('click', () => {
+      narrowNavMenu.classList.remove('showing');
+    });
+  }
+}
+
+/* global initializeSwipeGestures */
+/* global slideSidebar */
+/* global InstantClick */
```
Comment (rhymes) [inline app/assets/javascripts/initializers/initializeDrawerSliders.js]: these should be put at the top, you can simplify with `/* global initializeSwipeGestures, slideSidebar, InstantClick */`

File: app/assets/javascripts/initializers/initializeDrawerSliders.js [hunk 2]
```diff
@@ -1,57 +1,66 @@
+'use strict';
+
+function listenForNarrowMenuClick() {
+  var navLinks = document.getElementsByClassName('narrow-nav-menu');
+  var narrowFeedButt = document.getElementById('narrow-feed-butt');
+  var narrowNavMenu = document.getElementById('narrow-nav-menu');
```
Comment (rhymes) [inline app/assets/javascripts/initializers/initializeDrawerSliders.js]: something is odd here, the `navLinks` are an array of items with class `.narrow-nav-menu` and the `narrowNavMenu` is an item whose ID is `#narrow-nav-menu`. If I grep the code for `narrow-nav-menu` there's only one file containing that string - https://github.com/thepracticaldev/dev.to/blob/master/app/views/stories/_narrow_nav_menu.html.erb - which makes me think there's something odd going on. Why are we iterating on a list of one to remove a class from the same exact element? And then we add event handlers to add and remove the `showing` class.

I'm not 100% sure about what this function is for but it looks like me the code can be simplified.

Pinging @benhalpern who might know more about this

File: app/assets/javascripts/initializers/initializeDrawerSliders.js [hunk 3]
```diff
@@ -1,57 +1,66 @@
...
+    });
+  }
+}
+
+/* global initializeSwipeGestures */
+/* global slideSidebar */
+/* global InstantClick */
 function initializeDrawerSliders() {
-  if(!initializeSwipeGestures.called) {
-    swipeState = "middle";
```
Comment (rhymes) [inline app/assets/javascripts/initializers/initializeDrawerSliders.js]: this look important for gestures, the function that's called right after

File: app/assets/javascripts/initializers/initializeDrawerSliders.js [hunk 4]
```diff
@@ -1,57 +1,66 @@
...
+  if (!initializeSwipeGestures.called) {
     initializeSwipeGestures();
   }
-  if (document.getElementById("on-page-nav-controls")){
-    if (document.getElementById("sidebar-bg-left")){
-      document.getElementById("sidebar-bg-left").onclick = function(){
-        swipeState = "middle";
```
Comment (rhymes) [inline app/assets/javascripts/initializers/initializeDrawerSliders.js]: this too looks important, but I could be wrong

File: app/assets/javascripts/initializers/initializeDrawerSliders.js [hunk 5]
```diff
@@ -1,57 +1,66 @@
...
+    if (document.getElementById('sidebar-bg-left')) {
+      document
+        .getElementById('sidebar-bg-left')
+        .addEventListener('click', () => {
+          slideSidebar('left', 'outOfView');
+        });
     }
-    if (document.getElementById("sidebar-bg-right")){
-      document.getElementById("sidebar-bg-right").onclick = function(){
-        swipeState = "middle";
```
Comment (rhymes) [inline app/assets/javascripts/initializers/initializeDrawerSliders.js]: this too looks limportant

File: app/assets/javascripts/initializers/initializeDrawerSliders.js [hunk 6]
```diff
@@ -1,57 +1,66 @@
...
+      document
+        .getElementById('sidebar-bg-right')
+        .addEventListener('click', () => {
+          slideSidebar('right', 'outOfView');
+        });
     }
 
-    if (document.getElementById("on-page-nav-butt-left")){
-      document.getElementById("on-page-nav-butt-left").onclick = function(){
-        swipeState = "left"
```
Comment (rhymes) [inline app/assets/javascripts/initializers/initializeDrawerSliders.js]: this too looks important

File: app/assets/javascripts/initializers/initializeDrawerSliders.js [hunk 7]
```diff
@@ -1,57 +1,66 @@
...
+    if (document.getElementById('on-page-nav-butt-left')) {
+      document
+        .getElementById('on-page-nav-butt-left')
+        .addEventListener('click', () => {
+          slideSidebar('left', 'intoView');
+        });
     }
-    if (document.getElementById("on-page-nav-butt-right")){
-      document.getElementById("on-page-nav-butt-right").onclick = function(){
-        swipeState = "right"
```
Comment (rhymes) [inline app/assets/javascripts/initializers/initializeDrawerSliders.js]: this too looks important

**Comment by sagarchoudhary96** [comment]
@rhymes @benhalpern any updates on this ?

**Comment by sagarchoudhary96** [comment]
@rhymes @benhalpern deleted fork by mistake so created new PR for the same : #4404

**Comment by sagarchoudhary96** [comment]
@rhymes @benhalpern  i deleted the fork by mistake so created new PR again #4405

**Comment by rhymes** [comment]
Closing as duplicate of https://github.com/thepracticaldev/dev.to/pull/4405

**Pull request closed without merging**